pkg_resources/__init__.py

```diff
@@ -2201,7 +2296,7 @@ def resolve_egg_link(path):
 )
 
 
-def register_namespace_handler(importer_type, namespace_handler):
+def register_namespace_handler(importer_type: type, namespace_handler: _AdapterType):
```

Comment:
Instead of `_AdapterType`, it would be more "sound" here to have a callable like this:

```python
        def namespace_handler(importer, path_entry, moduleName, module):
            # return a path_entry to use for child packages
```

wouldn't it?

Reply:
See comment https://github.com/pypa/setuptools/pull/4246/files#r1596038268
This would become:
```py
def register_namespace_handler(
    importer_type: Type[_T],
    namespace_handler: _NSHandlerType[_T],
) -> None: ...
```